snuba_sdk/query_validation.py

```diff
@@ -86,15 +100,123 @@ def _validate_join(match: Join, all_columns: Set[Expression]) -> None:
             )
 
 
-def _validate_entity(match: Entity, all_columns: Set[Expression]) -> None:
+def validate_entity(match: Entity, all_columns: Set[Expression]) -> None:
     """
     Perform the checks to validate the match entity:
 
     Ensure that all the columns referenced in the query are in the data model for that entity.
 
     :param match: The Entity of the query.
+    :type match: Entity
     :param all_columns: All the columns referenced in the query.
+    :type all_columns: Set[Expression]
     """
     for column in all_columns:
         assert isinstance(column, Column)
         column.validate_data_model(match)
+
+
+class RequiredColumnError(Exception):
+    pass
+
+
+def _check_entity_required_columns_in_conditions(
+    entity: Entity, conditions: Optional[Sequence[Union[BooleanCondition, Condition]]]
+) -> None:
+    """
+    For a given entity, flatten the AND conditions to find just the top level ones and
```

Comment:
The `validate_required_columns` docstring gives almost all the context that this function does. It may be a good idea to consolidate the two under the public module function and just point to it from here. The example is good though so you should definitely keep that